fpga/src/main/scala/arty/System.scala

```diff
@@ -0,0 +1,51 @@
+// See LICENSE for license details.
+package sifive.freedom.everywhere.e300artydevkit
+
+import Chisel._
+
+import freechips.rocketchip.config._
+import freechips.rocketchip.subsystem._
+import freechips.rocketchip.devices.debug._
+import freechips.rocketchip.devices.tilelink._
+import freechips.rocketchip.diplomacy._
+import freechips.rocketchip.system._
+
+import sifive.blocks.devices.mockaon._
+import sifive.blocks.devices.gpio._
+import sifive.blocks.devices.pwm._
+import sifive.blocks.devices.spi._
+import sifive.blocks.devices.uart._
+import sifive.blocks.devices.i2c._
+
+//-------------------------------------------------------------------------
+// E300ArtyDevKitSystem
+//-------------------------------------------------------------------------
+
+class E300ArtyDevKitSystem(implicit p: Parameters) extends RocketSubsystem
```

Comment:
I think even this first step should be putting a chipyard based generator on the fpga.